sky/clouds/service_catalog/common.py

```diff
@@ -680,7 +680,7 @@ def get_image_id_from_tag_impl(df: 'pd.DataFrame', tag: str,
     df = _filter_region_zone(df, region, zone=None)
     assert len(df) <= 1, ('Multiple images found for tag '
                           f'{tag} in region {region}')
-    if len(df) == 0:
+    if not df:
```

Comment:
Not sure if we could do this to dfs. Pls also check other places in this PR

```bash
>>> df = pd.DataFrame()
>>> bool(df)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/txia/miniconda3/envs/sky-serve/lib/python3.9/site-packages/pandas/core/generic.py", line 1466, in __nonzero__
    raise ValueError(
ValueError: The truth value of a DataFrame is ambiguous. Use a.empty, a.bool(), a.item(), a.any() or a.all().
```

Reply:
Checked by
```bash
andyl@DESKTOP-7FP6SMO ~/skypilot (fix-serve-types)> git diff upstream/master --name-only | xargs grep -l "DataFrame"                                            (skypilot) 
sky/cli.py
sky/clouds/service_catalog/common.py
sky/clouds/service_catalog/data_fetchers/fetch_azure.py
sky/clouds/service_catalog/data_fetchers/fetch_vsphere.py
```